jwql/instrument_monitors/common_monitors/readnoise_monitor.py

```diff
@@ -0,0 +1,625 @@
+
+        amps : dict
+            Dictionary containing amp boundary coordinates (output from
+            ``amplifier_info`` function)
+            ``amps[key] = [(xmin, xmax, xstep), (ymin, ymax, ystep)]``
+
+        Returns
+        -------
+        amp_stats : dict
+            Contains the image statistics for each amp.
+        """
+
+        amp_stats = {}
+
+        for key in amps:
+            x_start, x_end, x_step = amps[key][0]
+            y_start, y_end, y_step = amps[key][1]
+
+            # Find sigma-clipped mean/stddev values for this amp
+            amp_data = image[y_start: y_end: y_step, x_start: x_end: x_step]
+            clipped = sigma_clip(amp_data, sigma=3.0, maxiters=5)
+            amp_stats['amp{}_mean'.format(key)] = np.nanmean(clipped)
+            amp_stats['amp{}_stddev'.format(key)] = np.nanstd(clipped)
+
+            # Find the histogram stats for this amp
+            n, bin_centers = self.make_histogram(amp_data)
+            amp_stats['amp{}_n'.format(key)] = n
+            amp_stats['amp{}_bin_centers'.format(key)] = bin_centers
+
+        return amp_stats
+
+    def get_metadata(self, filename):
+        """Collect basic metadata from a fits file.
+
+        Parameters
+        ----------
+        filename : str
+            Name of fits file to examine.
+        """
+
+        header = fits.getheader(filename)
+
+        try:
+            self.detector = header['DETECTOR']
+            self.read_pattern = header['READPATT']
+            self.subarray = header['SUBARRAY']
+            self.nints = header['NINTS']
+            self.ngroups = header['NGROUPS']
+            self.substrt1 = header['SUBSTRT1']
+            self.substrt2 = header['SUBSTRT2']
+            self.subsize1 = header['SUBSIZE1']
+            self.subsize2 = header['SUBSIZE2']
+            self.date_obs = header['DATE-OBS']
+            self.time_obs = header['TIME-OBS']
+            self.expstart = '{}T{}'.format(self.date_obs, self.time_obs)
+        except KeyError as e:
+            logging.error(e)
+
+    def identify_tables(self):
+        """Determine which database tables to use for a run of the readnoise
+        monitor.
+        """
+
+        mixed_case_name = JWST_INSTRUMENT_NAMES_MIXEDCASE[self.instrument]
+        self.query_table = eval('{}ReadnoiseQueryHistory'.format(mixed_case_name))
+        self.stats_table = eval('{}ReadnoiseStats'.format(mixed_case_name))
+
+    def image_to_png(self, image, outname):
+        """Outputs an image array into a png file.
+
+        Parameters
+        ----------
+        image : numpy.ndarray
+            2D image array.
+
+        outname : str
+            The name given to the output png file.
+
+        Returns
+        -------
+        output_filename : str
+            The full path to the output png file.
+        """
+
+        output_filename = os.path.join(self.data_dir, '{}.png'.format(outname))
+
+        # Get image scale limits
+        z = ZScaleInterval()
```

Comment:
I would suggest to rename this variable something like `zscale`.  It is good to avoid one-letter variables when you can.